samples/metrics/src/main/java/org/wso2/msf4j/sample/metrics/Application.java

```diff
@@ -17,19 +17,39 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.wso2.carbon.config.ConfigProviderFactory;
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.config.provider.ConfigProvider;
 import org.wso2.msf4j.MicroservicesRunner;
 import org.wso2.msf4j.analytics.metrics.MetricsInterceptor;
 import org.wso2.msf4j.sample.metrics.service.DemoService;
 
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.nio.file.Path;
+
 /**
  * Main Application Class.
  */
 public class Application {
 
     private static final Logger logger = LoggerFactory.getLogger(Application.class);
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws ConfigurationException, FileNotFoundException {
         logger.info("Starting the Microservice with Metrics");
+
+        // Configuration provider should be set
+        URL resource = Application.class.getClassLoader().getResource("metrics.yaml");
+        if (resource == null) {
+            throw new FileNotFoundException("Metrics configuration resource not found");
+        }
+
+        Path configPath = new File(resource.getFile()).toPath();
+        // We do not need secure vault for this sample and therefore it is null
+        ConfigProvider configProvider = ConfigProviderFactory.getConfigProvider(configPath, null);
```

Comment:
Shall we move this logic to org.wso2.msf4j.analytics.metrics.Metrics class. There we can check the configProvider and if not load the default one. and also allow use set configuration file in system property.